src/main/ml-modules/root/test/test-controller.xqy

```diff
@@ -28,46 +30,89 @@ declare function list()
 				else
-					let $uris := helper:list-from-database($db-id, $root, ())
+					let $uris := helper:list-from-database($db-id, $root, (), 'suites')
 					return
 						fn:distinct-values(
 							for $uri in $uris
 							let $path := fn:replace(cvt:basepath($uri), fn:concat($root, "test/suites/?"), "")
 							where $path ne "" and fn:not(fn:contains($path, "/")) and fn:not($path = $suite-ignore-list)
 							return
 								$path)
-			for $suite as xs:string in $suites
-			let $tests as xs:string* :=
+			let $main-formats as xs:string* :=
 				if ($db-id = 0) then
-					xdmp:filesystem-directory(fn:concat($root, $FS-PATH, "test/suites/", $suite))/dir:entry[dir:type = "file" and fn:not(dir:filename = $test-ignore-list)]/dir:filename[fn:ends-with(., ".xqy") or fn:ends-with(., ".sjs")]
+					xdmp:filesystem-directory(fn:concat($root, $FS-PATH, "test/formats"))/dir:entry[dir:type = "file" and fn:not(dir:filename = $test-ignore-list)]/dir:filename[fn:ends-with(., ".xsl")]
 				else
-					let $uris := helper:list-from-database(
-						$db-id, $root, fn:concat($suite, '/'))
+					let $uris := helper:list-from-database($db-id, $root, (), 'formats')
 					return
 						fn:distinct-values(
 							for $uri in $uris
-							let $path := fn:replace($uri, fn:concat($root, "test/suites/", $suite, "/"), "")
-							where $path ne "" and fn:not(fn:contains($path, "/")) and fn:not($path = $test-ignore-list) and (fn:ends-with($path, ".xqy") or fn:ends-with($path, ".sjs"))
+							let $path := fn:replace($uri, fn:concat($root, "test/formats/"), "")
+							where $path ne "" and fn:not(fn:contains($path, "/")) and fn:not($path = $test-ignore-list) and (fn:ends-with($path, ".xsl"))
 							return
 								$path)
-			where $tests
-			return
-				element t:suite {
-					attribute path {$suite},
-					element t:tests {
-						for $test in $tests
-						return
-							element t:test {
-								attribute path {$test}
-							}
-					}
-				}
+			return (
+			    for $suite as xs:string in $suites
+			    let $tests as xs:string* :=
+				    if ($db-id = 0) then
+				        xdmp:filesystem-directory(fn:concat($root, $FS-PATH, "test/suites/", $suite))/dir:entry[dir:type = "file" and fn:not(dir:filename = $test-ignore-list)]/dir:filename[fn:ends-with(., ".xqy") or fn:ends-with(., ".sjs")]
+				    else
+				        let $uris := helper:list-from-database(
+						    $db-id, $root, fn:concat($suite, '/'), 'suites')    
+					    return
+						    fn:distinct-values(
+						        for $uri in $uris
+						        let $path := fn:replace($uri, fn:concat($root, "test/suites/", $suite, "/"), "")
+						        where $path ne "" and fn:not(fn:contains($path, "/")) and fn:not($path = $test-ignore-list) and (fn:ends-with($path, ".xqy") or fn:ends-with($path, ".sjs"))
+						        return
+						            $path)
+			    let $formats as xs:string* :=
```

Comment:
`$main-formats` above let's us set up formats that can be used for project-wide test reporting. It looks like `$formats` here is intended to allow formatting specific to an individual suite. What's the use case for that? (Seems like an unnecessary complication to me.)